app/controllers/admin/applications_controller.rb

```diff
@@ -0,0 +1,22 @@
+class Admin::ApplicationsController < ApplicationController
+
+  def show
+    @application = Application.find(params[:id])
+
+    if params[:search].present?
+      @pets = Pet.search(params[:search])
+    end
+  end
+
+    def approve_app
+      @application = Application.find(params[:id])
+      @application.approve
+      redirect_to "/admin/applications/#{@application.id}"
+    end
+
+    def reject_app
+      @application = Application.find(params[:id])
+      @application.reject
+      redirect_to "/admin/applications/#{@application.id}"
+    end
+end
```

Comment:
For US 12, 13, 14 we are not approving or rejecting the application, we are approving or rejecting the specific pet on that application.